mlir/lib/Dialect/Rock/utility/loweringUtils.cpp

```diff
@@ -781,3 +782,66 @@ LogicalResult mlir::rock::checkLDSSize(StringAttr arch, int64_t ldsBytes) {
   const int64_t ldsSize = rock::lookupArchInfo(arch).maxSharedMemPerWG;
   return success(ldsBytes <= ldsSize);
 }
+
+// Trace an arg back to its alloc.
+static void traceGemmAllocToArgs(memref::AllocOp buffer,
+                                 const BufferDependencyAnalysis &deps,
+                                 SmallVector<BlockArgument> &args) {
+  IRRewriter rewriter(buffer.getContext());
+  std::optional<llvm::SmallVector<OpOperand *>> readersOperands =
+      deps.getReaders(buffer);
+  if (!readersOperands.has_value())
+    return;
+  for (OpOperand *readerOperand : readersOperands.value()) {
+    if (memref::CopyOp copyOp =
```

Comment:
why copy op special cased here ? 
I think we can still look for MemoryEffects where the operation's operand is being written && its a block argument. I can imagine its legit to provide kernels which directly writes to func args without a copy.

Reply:
you're right, I can change that.